src/llama_recipes/utils/train_utils.py

```diff
@@ -55,13 +58,15 @@ def train(model, train_dataloader,eval_dataloader, tokenizer, optimizer, lr_sche
     if train_config.use_fp16 and train_config.enable_fsdp:
         scaler = ShardedGradScaler()
     elif train_config.use_fp16 and not train_config.enable_fsdp:
-        scaler = torch.cuda.amp.GradScaler()
+        scaler = torch.xpu.amp.GradScaler() if is_xpu_available() else torch.cuda.amp.GradScaler()
```

Comment:
Do we really have torch.xpu.amp.GradScaler() already?

Reply:
Yes I also had a doubt regarding this, not sure we have xpu.amp.GradScaler() .

Reply:
> Do we really have torch.xpu.amp.GradScaler() already?

Thanks @gujinghui , @abhilash1910 , yes, we don't support torch.xpu.amp.GradScaler(), I will remove it and update it once we support this API.

Reply:
Should be yes? Assert to stop the workload with graceful exit message?

Reply:
> Should be yes? Assert to stop the workload with graceful exit message?

Thanks for the suggestion. We have refined the code again, and could you please help review again? Thanks!